### PR TITLE
Add Volunteer/Admin user models

### PR DESCRIPTION
**Pull request by peterghrong**
# Description

Add Volunteer/Admin User models,
I skipped api because there were a few things that I am not too sure about.

Questions:
1. User is missing a password field in the class diagram, wondering if I should add that.
2. Some routes should probably be admin access only, users probably shouldnt have access to the admin route, wondering what you think of that.
3. Need some more issues open to point directions : ) am lazy. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.  
Provide instructions so we can **reproduce**.  
Please also list any relevant details for your test configuration

- [x] Run the test command, passed 2 tests
Note: just testing models is quite redundant, usually a model creation should be coupled with an api route, I had to do this because there were a few things that I am unsure about the API. 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I will rebase before merging

### CODE REVIEW

**Comment by kehwhy** [comment]
Code looks good! I'll try to get another reviewer to check it as well. 

As for API:
- Yes, there should be a password field. Password will be provided to the user by MADA and may/may not be the same for all users but will definitely need to be sent and verified!
- You're right there will definitely be some admin-specific routes (off the top of my head /tasks should show ALL tasks for Admins and only assigned tasks for volunteers.) If we include a userID then we can check if the user is an admin to determine what information to send? And we can send an empty response or 404 when volunteers access admin routes? Volunteers will only be accessing via a mobile app and admins will be accessing via the web app, not sure if that can be leveraged in any way.

File: backend/app/models.py [hunk 1]
```diff
@@ -1 +1,48 @@
...
+
+
+class UserMixin(object):
+    id = db.Column(
+        UUID(as_uuid=True),
+        primary_key=True,
+        default=uuid.uuid4
+    )
+    name = db.Column(db.String(64), nullable=False)
+    phone_number = db.Column(db.Integer, nullable=False)
```
Comment (pengyk) [inline backend/app/models.py:14]: I think phone number should be String and not integer, its not really a number but a collection of digits I believe

Comment (pengyk) [reply backend/app/models.py:14]: Might overflow unless I forgot how to count

**Review by pengyk — CHANGES_REQUESTED** [review]
Tests pass for me too :)))

File: backend/app/models.py (hunk 1 above)
Comment (peterghrong) [reply backend/app/models.py:14]: Good point .

Comment (peterghrong) [reply backend/app/models.py]: APPROVE DADDY

**Review by pengyk — APPROVED** [review]

**Pull request merged**